components/bio-formats-tools/src/loci/formats/tools/ImageInfo.java

```diff
@@ -1034,7 +1034,14 @@ public boolean testRead(String[] args)
 
     // initialize reader
     long s = System.currentTimeMillis();
-    reader.setId(id);
+    try {
+      reader.setId(id);
+    } catch (FormatException exc) {
+      reader.close();
```

Comment:
No objections to this being here, but we should probably make it a bug for any reader to throw from `setId` **without** closing the underlying files. Not sure if FindBugs could help us with that.